### PR TITLE
feat: add the ability to read the port of a Spring Boot project

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

File: frameworks/spring-boot/src/main/java/io/ap4k/spring/SpringPropertiesHolder.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import java.util.concurrent.atomic.AtomicReference;
+
+public interface SpringPropertiesHolder extends WithProject {
+
+    AtomicReference<Map<String, Object>> springProperties = new AtomicReference<>(null);
+
+    default Map<String, Object> getSpringProperties() {
+        if (springProperties.get() == null) {
+            final Map<String, Object> properties = new HashMap<>();
+            properties.putAll(getProject().parseAppResourceFile("application.properties"));
```
Comment (iocanel) [inline frameworks/spring-boot/src/main/java/io/ap4k/spring/SpringPropertiesHolder.java]: I am wondering if we should also take profiles into consideration.

File: core/src/main/java/io/ap4k/project/Project.java [hunk 2]
```diff
@@ -74,18 +74,26 @@ public Project withAp4kOutputDir(String ap4kOutputDir) {
     return new Project(root, ap4kInputDir, ap4kOutputDir, buildInfo);
   }
 
-  public Map<String, Object> parseAppResourceFile(String resourceName) throws IOException {
```
Comment (iocanel) [inline core/src/main/java/io/ap4k/project/Project.java]: Let's just call it `parseResource` as it could actually parse any resource.

Comment (geoand) [reply core/src/main/java/io/ap4k/project/Project.java]: Fixed

File: frameworks/spring-boot/src/main/java/io/ap4k/spring/SpringPropertiesHolder.java (hunk 1 above)
Comment (geoand) [reply frameworks/spring-boot/src/main/java/io/ap4k/spring/SpringPropertiesHolder.java]: Sorry, I just got notified about this and hadn't seen it.
We could do that yes, *if* the profile is defined in the file and not by some other means (by in my experience that doesn't happen often).

**Pull request merged**